jgrapht-core/src/main/java/org/jgrapht/alg/connectivity/MedianGraph.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * (C) Copyright 2024-2024, by Louis Depiesse, Auguste Célérier and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+
+package org.jgrapht.alg.connectivity;
+
+import org.jgrapht.Graph;
+import org.jgrapht.alg.shortestpath.FloydWarshallShortestPaths;
+import org.jgrapht.alg.util.Pair;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Provides utility methods to determine if a graph is a median graph.
+ * 
+ * A median graph is a graph where, for any triplet of vertices (u, v, w),
+ * there exists exactly one vertex that minimizes the sum of distances to these three vertices.
+ */
+public class MedianGraph {
```

Comment:
This class name needs improvement.

Option 1:  rename it to GraphMedians to reflect our naming convention for utility classes, and change the class to abstract, since it should never be instantiated.

Option 2: change it to a non-static class GraphMedianInspector taking the graph as constructor parameter, in which case we can expose the isMedianGraph and findMedians methods as public and non-static (with reduced parameters).